grpclog/grpclog.go

```diff
@@ -218,3 +226,22 @@ func (l *GrpcLog) Marshal() ([]byte, error) {
 	}
 	return bytes, nil
 }
+
+type hasURN interface {
+	GetUrn() string
+}
+
+// FindByUrn finds the first entry with the given resource URN, or nil if none is found.
+func FindByUrn[TRequest resourceRequest, TResponse resourceResponse](entries []TypedEntry[TRequest, TResponse],
+	urn string) *TypedEntry[TRequest, TResponse] {
+	// nolint:copylocks
```

Comment:
Why is this needed?

Reply:
Gotcha. We should be able to avoid the copy (and the nolint) by skipping the assignment inside the loop:

```go
for idx := range entries {
    switch r := any(&entries[idx].Request).(type) {
    case hasURN:
        if r.GetUrn() == urn {
            return &entries[idx]
        }
    }
}
```